src/main/java/delete_post/drivers/DeletePostDataAccess.java

```diff
@@ -2,24 +2,120 @@
 
 import delete_post.use_case.DeletePostDsGateway;
 
+import com.opencsv.CSVReader;
+import com.opencsv.CSVWriter;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvException;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 public class DeletePostDataAccess implements DeletePostDsGateway{
 
-    private final String path;
+    private final String postPath;
+    private final String userPath;
 
-    public DeletePostDataAccess(String path){
-        this.path = path;
+    public DeletePostDataAccess(String postPath, String userPath){
+        this.postPath = postPath;
+        this.userPath = userPath;
     }
 
     @Override
-    public void removeFavourite(int postId, int userId){
-
+    public int getPostUser(int postId){
+        List<String[]> postData = readFile(postPath);
+        try {
+            for (String[] row : postData) {
+                if (Integer.parseInt(row[0]) == postId) {
+                    return Integer.parseInt(row[1]);
+                }
+            }
+        }
+        catch (Exception e){
+            System.out.println("Error: Could not find post in database.");
+        }
+        return -1;
     }
-    public void removeTag(int postId, String tag){
 
+    public void removeFavourites(int postId){
+        List<String> favUsers;
+        favUsers = new ArrayList<String>(Arrays.asList(getFavourite(postId).split(" ")));
+
+        List<String[]> userData = readFile(userPath);
+
+        for (String[] row: userData){
+            if (favUsers.contains(row[0])){
+                StringBuilder updateFav = new StringBuilder();
+                String[] oldFavList = row[8].split(" ");
+                for (String favId : oldFavList){
+                    if (Integer.parseInt(favId) != postId) {
+                        if (updateFav != null){
+                            updateFav.append(" ");
+                        }
+                        updateFav.append(favId);
+                    }
+                }
+                row[8] = updateFav.toString();
+            }
+        }
+        fileWriter(userPath, userData);
     }
     public void deletePost(int postId){
 
+        List<String[]> postData = readFile(postPath);
+
+        int rowNum = 0;
+        for (String[] row: postData){
```

Comment:
Modifying the length of a list while iterating over it is dangerous. You could instead make a new list and add all the posts except the one you want to remove.